src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java

```diff
@@ -84,6 +85,10 @@ protected void runTool( JavaSparkContext ctx ) {
             throw new UserException.Require2BitReferenceForBroadcast();
         }
 
+        if (joinStrategy == JoinStrategy.OVERLAPS_PARTITIONER && getHeaderForReads().getSortOrder() != SAMFileHeader.SortOrder.coordinate) {
+            throw new UserException.BadInput("Reads must be coordinate sorted when using the overlaps partitioner join strategy.");
+        }
```

Comment:
BQSR does not shuffle the reads so they will have the same sort order as the input, so I don't think a check is needed.